python/taichi/lang/snode.py

```diff
@@ -12,52 +12,119 @@
+
+        Args:
+            axis (List[Axis]): Axis to activate, must be 1.
+            dimension (int): Shape of the axis.
+            chunk_size (int): Chunk size.
+
+        Returns:
+            The added :class:`~taichi.lang.SNode` instance.
+        """
+        assert len(axis) == 1
         if chunk_size is None:
             chunk_size = dimension
-        return SNode(self.ptr.dynamic(index[0], dimension, chunk_size))
+        return SNode(self.ptr.dynamic(axis[0], dimension, chunk_size))
+
+    def bitmasked(self, axes, dimensions):
+        """Adds a bitmasked SNode as a child component of `self`.
+
+        Args:
+            axes (List[Axis]): Axes to activate.
+            dimensions (Union[List[int], int]): Shape of each axis.
 
-    def bitmasked(self, indices, dimensions):
+        Returns:
+            The added :class:`~taichi.lang.SNode` instance.
+        """
         if isinstance(dimensions, int):
-            dimensions = [dimensions] * len(indices)
-        return SNode(self.ptr.bitmasked(indices, dimensions))
+            dimensions = [dimensions] * len(axes)
+        return SNode(self.ptr.bitmasked(axes, dimensions))
 
     @deprecated('_bit_struct', 'bit_struct')
     def _bit_struct(self, num_bits):
         return self.bit_struct(num_bits)
 
-    def bit_struct(self, num_bits):
+    def bit_struct(self, num_bits: int):
+        """Adds a bit_struct SNode as a child component of `self`.
+
+        Args:
+            num_bits: Number of bits to use.
+
+        Returns:
+            The added :class:`~taichi.lang.SNode` instance.
+        """
         return SNode(self.ptr.bit_struct(num_bits))
 
     @deprecated('_bit_array', 'bit_array')
-    def _bit_array(self, indices, dimensions, num_bits):
-        return self.bit_array(indices, dimensions, num_bits)
+    def _bit_array(self, axes, dimensions, num_bits):
+        return self.bit_array(axes, dimensions, num_bits)
+
+    def bit_array(self, axes, dimensions, num_bits):
+        """Adds a bit_array SNode as a child component of `self`.
 
-    def bit_array(self, indices, dimensions, num_bits):
+        Args:
+            axes (List[Axis]): Axes to activate.
+            dimensions (Union[List[int], int]): Shape of each axis.
+            num_bits (int): Number of bits to use.
+
+        Returns:
+            The added :class:`~taichi.lang.SNode` instance.
+        """
         if isinstance(dimensions, int):
-            dimensions = [dimensions] * len(indices)
-        return SNode(self.ptr.bit_array(indices, dimensions, num_bits))
+            dimensions = [dimensions] * len(axes)
+        return SNode(self.ptr.bit_array(axes, dimensions, num_bits))
 
     def place(self, *args, offset=None, shared_exponent=False):
+        """Places a list of Taichi fields under the `self` container.
+
+        Args:
+            *args (List[ti.field]): A list of Taichi fields to place.
+            offsest (tupe[numbers.Number]): Offset of the field domain.
```

Comment:
It seems `offset` as an argument is `numbers.Number` than a `Tuple`?

Reply:
Actually it's a `Union[Number, tuple[Number]]` :)